unlock-app/src/components/interface/locks/CheckoutUrl/elements/LocksForm.tsx

```diff
@@ -303,11 +329,24 @@ export const LocksForm = ({
     setNetwork(network)
     setLockAddress(lockAddress)
     onRemoveFromList(lockAddress)
-    onAddLock(lockAddress, network!, name)
+    addLockMutation.mutateAsync({
+      lockAddress,
+      network,
+      name,
+    })
   }
 
+  useEffect(() => {
+    setRecurring(locks[lockAddress]?.recurringPayments ?? '')
+  }, [lockAddress, locks])
+
   return (
     <div className="flex flex-col gap-2">
+      {addLockMutation?.isLoading && (
+        <Placeholder.Root>
+          <Placeholder.Line className="p-2" />
+        </Placeholder.Root>
+      )}
```

Comment:
```suggestion
      {addLockMutation?.isLoading && (
        <Placeholder.Root>
          <Placeholder.Line className="p-2" />
        </Placeholder.Root>
      )}
```

when checking `async` for the `recurring` value, I added here loading to give feedback to the user